models/user.py

```diff
@@ -32,7 +32,10 @@ def has_role(cls, role):
 class UserModel(BaseModel):
     __tablename__ = "users"
 
+    __mapper_args__ = {"polymorphic_identity": "users", "polymorphic_on": "type"}
```

Comment:
Should the polymorphic identity here be `"user"` (singular)? All the others are singular, but this one is plural.
I'm not sure if it really matters or not, but it seems like it would be good to stick with on or the other.

Reply:
Good catch. I'll get that updated.